### ISSUE TITLE
Any reproduce of ViT-VQGAN?

### ISSUE THREAD

**Issue by Corleone-Huang** [lucidrains/parti-pytorch#4]
Is there anyone who successfully trained a Vit-VQGAN?     Since the reconstruction result of my trained ViT-VQGAN (using the implementation of this repo and others' implementation, and also my implementation) is all quite far worse than the CNN-based VQGAN.

**Comment by snarb** [comment]
For me results was disappointing also. I have tried to train it for 13+ hours on 4 a100 on random  32 by 32 crops if SINGLE  image. And results were blurred. It was even unable to overfit well enough.  I am curious if author have tried similar simple overfiting scenario

**Comment by Corleone-Huang** [comment]
The results are also very disappointing for me either. I have tried many different experimental settings for training Vit-VQGAN on the FFHQ dataset using 4 RTX-3090 GPUs on 256x256 resolution, but None achieves reasonable results. 

Also, I have emailed the authors of Vit-VQGAN about the source code but receive no response. I am really wondering how to train it.

**Comment by snarb** [comment]
When I see some paper implementation without results in Readme, 90% there is something wrong. And often you need to rewrite to fix it.  I think this vq gan implementation should  be tested for beginning to overfit well on small image, then to overfit well on small with dataset like cifar 10. Thats should be fast to train, compare to the sota and make adjustments. Most probably we should revisit vit gan pipeline. Try plain l2 loss for the beginning. And it at least should  be not worse than plain cnn autoencoder, and vq autoencoder,  than add perceptual loss, compare, add discriminatory and so on. Transformer need a lot of data to generalize, but it should overfit on small dataset easily with enough iterations. IMHO

**Comment by lucidrains** [comment]
> 

32x32 crops won't work well with ViT unless the patch sizes are down to 2x2 pixels (256 tokens)

**Comment by lucidrains** [comment]
> For me results were disappointing also for this and another one vit gan implementation. I have tried to train it for 13+ hours using 4 A100 powerfull GPU on random 32 by 32 crops of SINGLE 80 by 80 tiny image. And results were blurred. It was even unable to overfit well enough. Matbe if I will train for a week it will be better but not sure that it will be sharp. I am curious if author have tried similar simple overfiting scenario

@snarb  32x32 crops won't work well with ViT unless the patch sizes are down to 2x2 pixels (256 tokens)

**Comment by zhuqiangLu** [comment]
I hope I am not too late here. 
I am also having a similar problem; the generator can produce some blur reconstructions, but it never gets sharp images. I also notice that the discriminator loss gets stuck at 3.x very quickly. 

In fact, when I train the taming transformer VQGAN, the model does not generate sharp images neither, not until the discriminator is properly trained. So, I think reducing the weight of gradient penalty may help. I haven't tried it yet though.
